### DISCUSSION TITLE
General Idea

### DISCUSSION THREAD

**Discussion by ArcticOwl-Dev** [AllenCaine/DoNotAutoCutTrees#2] [Ideas]
@Shizen5
Sorry I had not much time the last weeks.

I have moved your mod idea to a new repository, because alextd has updated the original mod. 
When you have time, you can look into. 
I changed things according to your feedback, thanks. 
Save game related settings I have not included, I think it is too confusing and complex for the most players. And also I think you don't need it, only in very special situations.

**Comment by Shizen5** [comment]
Hi, thank you for the update!

Save game related settings are not needed if the mod become standalone.
The mod seems great, but still not perfect:
_You covered only half precepts, there is "Cutting Trees" and "Trees", both don't like when we cut trees. It seems you focused on the first one, so the mod don't work with only the second precept.
_A separated setting for Slaves could be better (like being able to don't care about slave mood, but don't have to think to toggle on the setting when we recruit tree lover pawn)

_Then a bonus bigger feature who would be greatly appreciated : a gizmo toggle on prisoner/slave/guest pawn, to being able to quickly choose if we want to care/don't care about the mood of THIS pawn. I will try to explain clearly:
- If our colony is not tree lover, and we have checked settings to care about prisoner/slave/guest, the gizmo will make the pawn ignored. If the pawn is the only prisoner/slave/guest tree lover, the pawns will still auto-cut.
- If our colony is not tree lover, and we have unchecked settings to care about prisoner/slave/guest, the gizmo will make the pawn override the settings, so pawns will don't auto-cut. And when this pawn leave the map, the pawns will start auto-cut again, without having to touch settings.

Why this feature would be good? We don't care about mood of prisoners here to be sold, eaten, harvested, converted, but we do when we try to recruit them (or guard them for quest). We don't care about the mood of crappy temporary slaves, but we do when they are good with essential skill (medic). We don't really care about mood of poor refugees guests, but we do when it's royal guests.
An easier alternative way to do would be to put toggles in right bottom corner of the screen as I said before (we should be able to change settings during game quickly without having to go in mod setting panel), but Gizmo would be better :)

**Reply by ArcticOwl-Dev** [reply]
Hey Shizen5,

have a look at the newest release https://github.com/AllenCaine/DoNotAutoCutTrees/releases/tag/v1.1   

I think I have now implemented everything you suggested.  
let me know if you still have something to complain about. :laughing:

**Comment by Shizen5** [comment]
Hi, thank you for the update!
Don't worry, I'm french, I can always find something to complain about :D

So **first**, I got an error the first time I select a pet (horse, cat), I need to restart the game to have the error again:
```
System.NullReferenceException: Object reference not set to an instance of an object
  at DoNotAutoCutTrees.PlantUtility.CheckTreeMoodDebuff (Verse.Pawn pawn) [0x00006] in <f6ffc6b0bda34400a5d69837c72dcfe2>:0 
  at DoNotAutoCutTrees.PlaySettingsPatch+ManuellPawnListGizmo.Postfix (System.Collections.Generic.IEnumerable`1[Verse.Gizmo]& __result, Verse.Pawn __instance) [0x00014] in <f6ffc6b0bda34400a5d69837c72dcfe2>:0 
  at (wrapper dynamic-method) Verse.Pawn.Verse.Pawn.GetGizmos_Patch3(Verse.Pawn)
  at RimWorld.InspectGizmoGrid.DrawInspectGizmoGridFor (System.Collections.Generic.IEnumerable`1[T] selectedObjects, Verse.Gizmo& mouseoverGizmo) [0x0003a] in <b64badbf3c3d41018b3ca5d3e8c77771>:0 
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.Log:ErrorOnce (string,int)
RimWorld.InspectGizmoGrid:DrawInspectGizmoGridFor (System.Collections.Generic.IEnumerable`1<object>,Verse.Gizmo&)
RimWorld.MainTabWindow_Inspect:DrawInspectGizmos ()
RimWorld.InspectPaneUtility:ExtraOnGUI (RimWorld.IInspectPane)
RimWorld.MainTabWindow_Inspect:ExtraOnGUI ()
Verse.WindowStack:WindowStackOnGUI ()
RimWorld.UIRoot_Play:UIRootOnGUI ()
(wrapper dynamic-method) Verse.Root:Verse.Root.OnGUI_Patch1 (Verse.Root)
```
**Second**, there is an issue with the "Guests" check, when checked I am unable to auto-cut tree, without having guest. It happens when I have a slave with tree ideo (and when Slave is unchecked, as him gizmo).


**Third**, would be more a minor improvement:
The Gizmo only allow to Disable Autocut (when global toggle for this pawn is off), so the Gizmo should not appear when the global toggle is on (other way to say it: if the Gizmo will have no effect, it should stay hidden). Right now it's confusing.


Also, the Gizmo appears (and works) for visitors, I don't know if it's wanted, I don't know if someone will need it.

Without the first small error, your mod is already perfect for me (Colonist & manual checked do the job greatly).

**Reply by ArcticOwl-Dev** [reply]
good, i was worried i have nothing to do any more 😆

**first** and **third** should be solved. 
**second** i am not sure what are you mean with that. Can you explain it more in detail?

https://github.com/AllenCaine/DoNotAutoCutTrees/releases/tag/v1.2

**Reply by Shizen5** [reply]
Sorry, I will try to make it understandable ^^'
if I check "guests" in setting, and if at the same time I have a slave with tree lover precept in my colony, then my pawns will never auto-cut trees, even if "Slaves" in setting is uncheck, the slave Gizmo is unchecked too, and that I have no guest in my colony.
It seems that your mod think that the slave is a guest.

I will try 1.2 (or 1.3?) tomorrow :)

**Reply by Shizen5** [reply]
Thank you, the 1.2 is now part of my mod list :)

The **second** issue is still true, the thing is that Slaves are Guests (but Guests are not Slaves): When Guests setting is checked, the Gizmo for slave is hidden.

I'm waiting a release on Steam workshop!

---

The Gizmo appears also on enemies ^^

**Reply by ArcticOwl-Dev** [reply]
So, the problem with guest and slaves should be fixed. 
[The problem was, that slaves are also 'free colonists' and not assigned to your 'Homefaction' and that was my check for guest, the game has no simple function to get guest colonists]
Mod is published on Steam: https://steamcommunity.com/sharedfiles/filedetails/?id=2635266578

Gizmo: 
Yeah the pawn is your enemy, but you don't have to cut trees in front of him. For some reason there already mad at you, you should not provoke them even more. :satisfied:

seriously, i intentionally don't removed the gizmo from enemies and visitors. 
When you looking for new recruits, you can directly set the gizmo and you dont have to wait to in prison the pawn. 
Also when the enemies prepare for a time, you have the option to disable automatically cutting of trees for the pawns you want to capture. Also for visitors you want to capture, but you want to wait a half day until you capture them, you have the option to care about the mood of this future pawn of you  :grin:

**Reply by Shizen5** [reply]
Nice! Thank you again :)

Gizmo on enemies is not bad, I quickly know their ideology, and can toggle it on for downed enemies during caravan raids.

If you are bored, I have other mod idea, less essential, like changing the drum from Ideology into a true music instrument like the harp/piano from Royalty, without break the "drum party" ritual (I tried and fail); or an improvement of auto-clean feature from Common Sense (who make pawns act so stupidly sometime/often); or also turn auto-slaughter into auto-release for animal lovers.
